### DISCUSSION TITLE
How to force auto empty between rooms

### DISCUSSION THREAD

**Discussion by mclouse93** [Tasshack/dreame-vacuum#158] [Q&A]
So, I'm trying to get my dreametech Z10 Pro to vacuum one room, go back to the dock and auto empty, and start the next room. I created a script that executes the vacuuming of the first room, but it never moves on to the second room after waiting for the auto empty cycle to change from 1 (active) to 0 (idle). See the code below. Note that I the room numbers don't make sense. I guess it's from merging/splitting rooms. Anyone know if i can use a verbal name vs the room number?

Do I actually have to set a timer on how long to wait? My interpretation was that it would continue through the code and execute the next call after the auto empty status changed from active to idle.

alias: Vacuum Home with Emptying
sequence:
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 2
    target:
      entity_id: vacuum.dreamebot_z10_pro
  - wait_for_trigger:
      - platform: state
        entity_id:
          - sensor.dreamebot_z10_pro_auto_empty_status
        attribute: value
        from: "1"
        to: "0"
        for:
          hours: 0
          minutes: 0
          seconds: 0
    timeout:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 0
    continue_on_timeout: false
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 1
    target:
      entity_id: vacuum.dreamebot_z10_pro
  - wait_for_trigger:
      - platform: state
        entity_id:
          - sensor.dreamebot_z10_pro_auto_empty_status
        attribute: value
        from: "1"
        to: "0"
        for:
          hours: 0
          minutes: 0
          seconds: 0
    continue_on_timeout: true
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 10
    target:
      entity_id: vacuum.dreamebot_z10_pro
  - wait_for_trigger:
      - platform: state
        entity_id:
          - sensor.dreamebot_z10_pro_auto_empty_status
        attribute: value
        from: "1"
        to: "0"
    continue_on_timeout: true
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 6
    target:
      entity_id: vacuum.dreamebot_z10_pro
  - wait_for_trigger:
      - platform: state
        entity_id:
          - sensor.dreamebot_z10_pro_auto_empty_status
        attribute: value
        from: "1"
        to: "0"
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 9
    target:
      entity_id: vacuum.dreamebot_z10_pro
  - wait_for_trigger:
      - platform: state
        entity_id:
          - sensor.dreamebot_z10_pro_auto_empty_status
        attribute: value
        from: "1"
        to: "0"
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 8
    target:
      entity_id: vacuum.dreamebot_z10_pro
  - wait_for_trigger:
      - platform: state
        entity_id:
          - sensor.dreamebot_z10_pro_auto_empty_status
        attribute: value
        from: "1"
        to: "0"
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 4
    target:
      entity_id: vacuum.dreamebot_z10_pro
mode: single

**Comment by Tasshack** [comment]
Can you check that from the recorder that auto empty status entity state is actually changed? Since auto emptying ony takes 10 seconds, integration may not be quick enough to actually get the changed value.
You can try to listen [task status event](https://github.com/Tasshack/dreame-vacuum/blob/master/docs/events.md#dreame_vacuum_task_status) for detecting the cleanup complete with a little bit of delay after for vacuum to finish auto emptying before starting a new job.

**Reply by mclouse93** [reply]
appears it never saw the status change. 

(script.vacuum_dreame_z10_pro) turned on
Dreame Vacuum: DreameBot Z10 Pro
(vacuum.dreamebot_z10_pro) turned cleaning
Wait for When Value of DreameBot Z10 Pro Auto Empty Status changes from '1' to '0'
(script.vacuum_dreame_z10_pro) turned off
Aborted at July 5, 2023 at 12:53:11 PM (runtime: 0.73 seconds)

Executed: July 5, 2023 at 12:53:11 PM
Result:
wait:
  remaining: 0
  trigger: null
timeout: true


If I add a timer of like 5 minutes, would that solve the issue?

**Reply by Tasshack** [reply]
As i said, if the integration does not sees the value is changing you cannot use auto empty status for detecting a cleanup complete, you need to use the already implemented mechanism for it which is the HA events. Or alternatively you can try to check the `task_status` entity changes.
